events/views.py

```diff
@@ -4,13 +4,46 @@
 from django.urls import reverse
 from .models import Event, Location
 from django.contrib.auth.decorators import login_required
+from django.utils import timezone
+from .forms import EventFilterForm
 
 # Create your views here.
 
 
 def index(request):
-    events = Event.objects.filter(is_active=True)
-    return render(request, "events/events.html", {"events": events})
+    events = Event.objects.filter(end_time__gt=timezone.now(), is_active=True).order_by(
+        "-start_time"
+    )
+
+    # Process the filter form
+    if request.GET:
+        form = EventFilterForm(request.GET)
+        if form.is_valid():
+            if form.cleaned_data["start_time"]:
+                events = events.filter(start_time__gte=form.cleaned_data["start_time"])
+            if form.cleaned_data["end_time"]:
+                if form.cleaned_data["end_time"] <= timezone.now():
+                    # Return an error message
+                    # Handle this appropriately in your template
+                    return render(
+                        request,
+                        "events/events.html",
+                        {"error": "End time cannot be in the past."},
+                    )
```

Comment:
The filter works great! Thank you! Just one note here: there should be another kind of error message: if the end time filter is earlier than the start time filter, the user should be alerted that there's a mistake in the input. (#61)

Reply:
added checks for start time cannot be less than current time and end time cannot be less than start time


![image](https://github.com/gcivil-nyu-org/Wednesday-Fall2023-Team-3/assets/127150757/37024984-e5d1-43c3-83fa-6811222237d1)